### ISSUE TITLE
[Accessibility]: There is no tabIndex in ibm-search component

### ISSUE THREAD

**Issue by yashgarg1719** [IBM/carbon-components-angular#2188]
## Detailed description

> Describe in detail the issue you're having.

A keyboard-focusable (tabbable) element has been found with an invalid role. When users reach elements by tabbing with the keyboard, the element's role provides context about the expected interaction, and assistive technologies like screen readers can provide additional information about ways to interact. The only elements that should be in the tab order should be operable widgets. Inoperable elements, such as headings, can be given focus by developers; however, they should not be tabbable by the user.

It's likely that either a non-interactive element was given a tabIndex value of 0 or greater, or assigned an element’s role incorrectly. The Checker detects implied roles of interactive HTML elements that automatically get keyboard focus. The Checker also detects content blocks that get keyboard focus to enable scrolling of the area using arrow keys when coding CSS overflow: scroll or overflow: auto. Both of these situations are valid. However, if the element should not be operable with the keyboard, then it should not be tabbable and get keyboard focus.

Note: IBM considers this a violation of WCAG Success Criteria. Others may use a strict interpretation of the criterion and may only recommend this to be fixed. However, there is broad agreement that putting inoperable elements in the tab order has an accessibility impact on users. 

If users should not be able to interact with this element using the keyboard, replace the non-widget role, and/or use the explicit tabindex="-1" attribute to remove the element from the keyboard focus (tab) order. Or,if users should be able to interact with this element using the keyboard, use a valid widget role

This first example is a simple focusable toggle button. (JavaScript and CSS would need to be added to provide the appearance and full button functionality):
`<div id="saveChanges" tabindex="0" role="button" aria-pressed="false">Save</div>`

The second example is placed in the tab order automatically and provides features by default when using the HTML <button> element of type=”button” that informs the user about its intended interaction:
`<button type="button" id="saveChanges">Save</button>`

The third example shows how a block of text can get keyboard focus to enable scrolling of the area by specifying overflow: scroll in the CSS:
`<style> div.scroll { width: 200px; height: 100px; border: 2px solid black; overflow: scroll;} </style> ... <div class="scroll" tabindex="0">`

You can use the CSS overflow property when you want to have better control of the layout. The overflow property specifies horizontal and vertical scroll bars operable with arrow keys if content overflows an element's box.</div>

> Is this a feature request (new component, new icon), a bug, or a general issue?

Bug

> Is this issue related to a specific component?

`<ibm-search>

> What did you expect to happen? What happened instead? What would you like to see changed?

People using a screen reader, including blind, low vision and neurodivergent people People with low vision who use screen magnification with the keyboard People using voice control or a keyboard-like switch device People who rely on keyboard control

> What browser are you working in?

Firefox 101.0(64-bit)

> What version of the Carbon Design System are you using?

`carbon-components@10.5.0`
`carbon-components-angular@4.56.4`

> What WCAG 2.1 checkpoint does the issue violate?

4.1.2 Name, Role, Value ARIA specification - Widget role definition CSS overflow: Unit test - check tab order Unit test - confirm component keyboard interaction

## Additional information

* Screenshots or code

![image](https://user-images.githubusercontent.com/108860673/178151342-f7f30810-c094-43c5-ab69-a9d18953d46f.png)

**Comment by Akshat55** [comment]
Issue was resolved with PR #2242
